vignettes/rhino-fluent.Rmd

```diff
@@ -0,0 +1,681 @@
+
+So, this is how the app looks and works now!
+
+```{r, echo=FALSE}
+knitr::include_graphics("images/rhino_fluent_gif_1.gif")
+```
+
+## Improve visuals and layout for the table with elevated cards
+
+We can create cards with elevations just like in Fluent UI [documentation](https://developer.microsoft.com/en-us/fluentui#/styles/web/elevation), as we partially covered it above. For creating cards, we can use [`Stack`](https://appsilon.github.io/shiny.fluent/reference/Stack.html) from `shiny.fluent`.
+
+We will also style the data table area. The following code does the following - 
+
+* Confines the height and width of the table area.
+* Creates an elevated card around the data table area.
+
+```
+# app/view/datatable.R
+
+box::use(
+  dplyr[filter],
+  shiny[div, moduleServer, NS, reactive, renderUI, span, uiOutput],
+  shiny.fluent[DetailsList, Stack, Text, Toggle.shinyInput],
+  tibble[tibble],
+)
+
+columns <- tibble(
+  fieldName = c("rep_name", "date", "deal_amount", "city", "is_closed"),
+  name = c("Sales Rep", "Date", "Deal Amount", "City", "Is Closed?")
+)
+
+#' @export
+ui <- function(id) {
+  ns <- NS(id)
+  div(
+    Text("A randomly generated dataset of imaginary sales deals", variant = "large"),
+    div(
+      class = "ms-depth-4",
+      Toggle.shinyInput(ns("includeOpen"), label = "Include open deals")
+    ),
+    Stack(
+      horizontal = TRUE,
+      tokens = list(padding = 20, childrenGap = 10),
+      span(class = "ms-depth-8", uiOutput(ns("datatable")))
+    )
+  )
+}
+
+#' @export
+server <- function(id) {
+  moduleServer(id, function(input, output, session) {
+    filtered_deals <- reactive({
+      shiny.fluent::fluentSalesDeals |> filter(
+        is_closed | input$includeOpen
+      )
+    })
+    output$datatable <- renderUI({
+      DetailsList(items = filtered_deals(), columns = columns)
+    })
+  })
+}
+```
+The `Stack` element in the code allows for stacking multiple elements vertically or horizontally.
+
+The horizontal argument is set to TRUE to horizontally stack the two elements. The tokens argument is a list that specifies the padding and the gap between the two elements. In this case, the padding is set to 20 and the childrenGap is set to 10. This will add a padding of 20 pixels and a gap of 10 pixels between the other `span` of the `Stack` element.
+
+The span element has a class attribute set to `ms-depth-8`. This is a class provided by the Microsoft fluent UI that adds a shadow effect to the element, giving it a depth effect. The `uiOutput` element is used to render the `DetailsList` element that contains the filtered data.
+
+The additional style for the class `ms-depth-8` is added in the `app/style/main.scss` file. So it now is modified to -
+```
+// app/style/main.scss
+
+.ms-depth-4 {
+    display: flex; 
+    flex-wrap: wrap;
+    padding: 5px;
+}
+
+.ms-depth-8 {
+    height: 500px; 
+    overflow: auto; 
```

Comment:
This results in the resize of the table that happens after everything is rendered. Changing to `overflow: overlay;` should help. Unfortunately, you will need also to regenerate gifs.